src/main/java/edu/harvard/iq/dataverse/search/IndexServiceBean.java

```diff
@@ -1021,10 +1030,18 @@ private String addOrUpdateDataset(IndexableDataset indexableDataset) {
             return ex.toString();
         }
 
-        dvObjectService.updateContentIndexTime(dataset);
+        Long dsId = dataset.getId();
+        ///Dataset updatedDataset = (Dataset)dvObjectService.updateContentIndexTime(dataset);
+        ///updatedDataset = null;
+        // instead of making a call to dvObjectService, let's try and 
+        // modify the index time stamp using the local EntityManager:
+        DvObject dvObjectToModify = em.find(DvObject.class, dsId);
+        dvObjectToModify.setIndexTime(new Timestamp(new Date().getTime()));
+        dvObjectToModify = em.merge(dvObjectToModify);
+        dvObjectToModify = null;
 
```

Comment:
Since the call to `dvObjectService.updateContentIndexTime(dataset)` was removed should the call to `dvObjectService.updateContentIndexTime(dataverse)` (line 253 as of 0f814b6) be removed as well?

Reply:
This call to a remote service bean does not appear to be causing any trouble in indexDataverse(). To be preciese, indexDataverse() was not the source of any of that growing, and un-garbage-collectable heap. (I'm assuming because indexDataverse is a relatively shallow operation - it doesn't need to explore any cascading data structures nearly as much?)
To be honest, I'm not 100% positive it's needed in indexDataset() either... I just figured I would try and make *everything* happen inside the transaction, when this is called via indexDatasetInNewTransaction()

Reply:
Ok, no worries. Mostly I was wondering if `dvObjectService.updateContentIndexTime` could be safely deleted and I found that it was still being called.